trustpoint/sysconf/views.py

```diff
@@ -16,6 +21,79 @@
 from .models import LoggingConfig, NetworkConfig, NTPConfig, SecurityConfig
 
 
+class SecurityLevelMixin:
+    """A mixin that provides security featrues checks for Django views."""
+
+    _thread_locals = threading.local()
+
+    def __init__(self, security_feature: SecurityFeatures=None, no_permisson_url=None, *args, **kwargs) -> None:
+        """Initializes the SecurityLevelMixin with the specified security feature and redirect URL.
+
+        Parameters:
+        -----------
+        security_feature : SecurityFeatures, optional
+            The feature to check against the current security level (default is None).
+        no_permisson_url : str, optional
+            The URL to which the user is redirected if the feature is not allowed (default is None).
+        *args, **kwargs:
+            Additional arguments passed to the superclass initializer.
+        """
+        super().__init__(*args, **kwargs)
+        self.sec = SecurityManager()
+        self.security_feature = security_feature
+        self.no_permisson_url = no_permisson_url
+
+    @classmethod
+    def _get_security_level_instance(cls):
+        """Retrieves the current security level instance from the database, caching it in thread-local storage.
+
+        Returns:
+        --------
+        SecurityConfig
+            The current security level instance.
+        """
+        if not hasattr(cls._thread_locals, 'security_level_instance'):
+            cls._thread_locals.security_level_instance = SecurityConfig.objects.first()
+        return cls._thread_locals.security_level_instance
+
+    @classmethod
+    def get_security_level(cls):
+        """Returns the security mode of the current security level instance.
+
+        Returns:
+        --------
+        str
+            The security mode of the current security level instance.
+        """
+        return cls._get_security_level_instance().security_mode
+
+    @classmethod
+    def refresh_security_level_instance(cls):
+        """Refreshes the cached security level instance by reloading it from the database."""
+        cls._thread_locals.security_level_instance = SecurityConfig.objects.first()
```

Comment:
Doing this using threading locals seems like a good way to implement a singleton, as long as we can guarantee that all views are called on the same thread. An alternative might be https://python-patterns.guide/gang-of-four/singleton/ , however that seems to not work well with `__init__()`, which may or may not be required for Mixins.